src/builtin.c

```diff
@@ -548,6 +564,38 @@ static jv f_format(jq_state *jq, jv input, jv fmt) {
     }
     jv_free(input);
     return line;
+  } else if (!strcmp(fmt_s, "base64d")) {
+    jv_free(fmt);
+    input = f_tostring(jq, input);
+    const unsigned char* data = (const unsigned char*)jv_string_value(input);
+    int len = jv_string_length_bytes(jv_copy(input));
+    char result[(3 * len) / 4]; // decoded result buffer
+    int ri = 0;
+    int input_bytes_read=0;
+    uint32_t code = 0;
+    for (int i=0; i<len && data[i] != '='; i++) {
+      code <<= 6;
+      code |= BASE64_DECODE_TABLE[data[i]];
+      input_bytes_read++;
+
+      if (input_bytes_read == 4) {
+        result[ri++] = (code >> 16) & 0xFF;
+        result[ri++] = (code >> 8) & 0xFF;
+        result[ri++] = code & 0xFF;
+        input_bytes_read = 0;
+        code = 0;
+      }
+    }
+    if (input_bytes_read == 3) {
+      result[ri++] = (code >> 10) & 0xFF;
+      result[ri++] = (code >> 2) & 0xFF;
+    } else if (input_bytes_read == 2) {
+      result[ri++] = (code >> 4) & 0xFF;
+    }
+    jv line = jv_string("");
+    line = jv_string_append_buf(line, result, ri);
```

Comment:
Just use jv_string_sized().

Now, what if the decoded string is not UTF-8?  We won't get an error.  Instead we'll get a string with non-UTF-8 sequences replaced with `0xFFFD (REPLACEMENT CHARACTER)`.  I'd rather get an error but I know that will be harder, so we can leave that for some other day.